src/crypto/CHIPCryptoPAL.h

```diff
@@ -228,68 +228,120 @@ class ECPKey
                                                      const Sig & signature) const                                              = 0;
 };
 
-template <size_t Cap>
-class CapacityBoundBuffer
+template <size_t kCapacity>
+class SensitiveDataBuffer
 {
 public:
-    ~CapacityBoundBuffer()
+    ~SensitiveDataBuffer()
     {
         // Sanitize after use
-        ClearSecretData(&bytes[0], Cap);
+        ClearSecretData(mBytes);
     }
 
-    CapacityBoundBuffer & operator=(const CapacityBoundBuffer & other)
+    SensitiveDataBuffer & operator=(const SensitiveDataBuffer & other)
     {
         // Guard self assignment
         if (this == &other)
             return *this;
 
-        ClearSecretData(&bytes[0], Cap);
+        ClearSecretData(mBytes);
         SetLength(other.Length());
-        ::memcpy(Bytes(), other.Bytes(), other.Length());
+        ::memcpy(Bytes(), other.ConstBytes(), other.Length());
         return *this;
     }
 
-    /** @brief Set current length of the buffer that's being used
-     * @return Returns error if new length is > capacity
-     **/
-    CHIP_ERROR SetLength(size_t len)
+    /**
+     * @brief Set current length of the buffer
+     * @return Error if new length is exceeds capacity of the buffer
+     */
+    CHIP_ERROR SetLength(size_t length)
     {
-        VerifyOrReturnError(len <= sizeof(bytes), CHIP_ERROR_INVALID_ARGUMENT);
-        length = len;
+        VerifyOrReturnError(length <= kCapacity, CHIP_ERROR_INVALID_ARGUMENT);
+        mLength = length;
         return CHIP_NO_ERROR;
     }
 
-    /** @brief Returns current length of the buffer that's being used
-     * @return Returns 0 if SetLength() was never called
-     **/
-    size_t Length() const { return length; }
+    /**
+     * @brief Returns current length of the buffer
+     */
+    size_t Length() const { return mLength; }
 
-    /** @brief Returns max capacity of the buffer
-     **/
-    static constexpr size_t Capacity() { return sizeof(bytes); }
+    /**
+     * @brief Returns non-const pointer to start of the underlying buffer
+     */
+    uint8_t * Bytes() { return &mBytes[0]; }
 
-    /** @brief Returns pointer to start of underlying buffer
-     **/
-    uint8_t * Bytes() { return &bytes[0]; }
+    /**
+     * @brief Returns const pointer to start of the underlying buffer
+     */
+    const uint8_t * ConstBytes() const { return &mBytes[0]; }
 
-    /** @brief Returns const pointer to start of underlying buffer
-     **/
-    const uint8_t * ConstBytes() const { return &bytes[0]; }
+    /**
+     * @brief Constructs fixed span from the underlying buffer
```

Comment:
This is not actually constructing a fixed span....

Reply:
Good catch! I swapped the comments.